services/telemetry/config.go

```diff
@@ -0,0 +1,313 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Package telemetry provides telemetry functionality.
+package telemetry
+
+import (
+	_ "embed" //nolint:golint
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"aead.dev/minisign"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v3"
+)
+
+// ServiceConfig telemetry config.
+type ServiceConfig struct {
+	l                          *logrus.Entry
+	Enabled                    bool   `yaml:"enabled"`
+	LoadDefaults               bool   `yaml:"load_defaults"`                //nolint:tagliatelle
+	ConfigLocation             string `yaml:"config_location"`              //nolint:tagliatelle
+	DisableSigningVerification bool   `yaml:"disable_signing_verification"` //nolint:tagliatelle
+	Signing                    struct {
+		TrustedPublicKeys       []string             `yaml:"trusted_public_keys"` //nolint:tagliatelle
+		trustedPublicKeysParsed []minisign.PublicKey `yaml:"-"`
```

Comment:
with `Unmarshaler` interface, you will need to duplicate struct and it will involve a lot of boilerplate. Here is an [example](https://github.com/percona-platform/grafana/blob/45eb4a0adf663976269b133626d196fbd1528a75/pkg/services/ngalert/api/tooling/definitions/alertmanager.go#L478).
Additionally, you are limited with what is passed, you cannot have logger, for example.
In this case, initialization is an explicit call - `Init`.

I have checked, time.Duration can be deserialized with now. Simplified impl.

Reply:
I meant this: 
```
{
...
TrustedPublicKeys PubKeys `yaml:"trusted_public_keys"`
...
}

type PubKeys []minisign.PublicKey

func (pk PubKeys) UnmarshalYAML(value *Node) error {
// Unmarshall logic
}

```


BTW, let's not do that now.